### PR TITLE
[WIP] JS tests on 2.0 — DO NOT MERGE YET

### PR DESCRIPTION
**Pull request by janl**

### CODE REVIEW

**Comment by sebastianrothbucher** [comment]
I got a little further (up to "r" ;-) ). Still have to dig a little more into your work. Maybe we can copy over missing parts from my branch (didn't jet submit a PR anyway): https://github.com/apache/couchdb/compare/master...sebastianrothbucher:clustertest

File: test/javascript/tests/changes.js [hunk 1]
```diff
@@ -638,29 +664,36 @@ couchTests.changes = function(debug) {
...
   db.save({"bop" : "foom"});
   db.save({"bop" : "foom"});
+  // because of clustering, we need the 2nd entry as since value
+  req = CouchDB.request("GET", "/" + db_name + "/_changes");
 
   // simulate an EventSource request with a Last-Event-ID header
-  req = CouchDB.request("GET", "/test_suite_db/_changes?feed=eventsource&timeout=0&since=0",
-        {"headers": {"Accept": "text/event-stream", "Last-Event-ID": "2"}});
+  req = CouchDB.request("GET", "/" + db_name + "/_changes?feed=eventsource&timeout=0&since=0",
+        {"headers": {"Accept": "text/event-stream", "Last-Event-ID": JSON.stringify(JSON.parse(req.responseText).results[1].seq)}});
```
Comment (kxepal) [inline test/javascript/tests/changes.js]: See https://github.com/apache/couchdb-chttpd/pull/89 . I made this value completely opaque as it was before, so there is no need to JSON encode seq for Last-Event-ID header. However, we can discuss this moment.

Comment (sebastianrothbucher) [inline test/javascript/tests/changes.js]: did a PR: https://github.com/janl/couchdb/pull/4

**Comment by wohali** [comment]
@janl I think this is outdated, can we close?

**Comment by sebastianrothbucher** [comment]
as far as I can remember, we can indeed close it (leaving it up 2 @janl still)

**Pull request closed without merging**